app/uk/gov/hmrc/agentpermissions/service/TaxGroupsService.scala

```diff
@@ -117,20 +117,36 @@ class TaxGroupsServiceImpl @Inject() (
       existingTaxGroups <- getAllTaxServiceGroups(arn)
 
       taxServiceIds = existingTaxGroups.map(groups => groups.service)
-      combinedCount = fullCount.fold(Map.empty[String, Int])(fc => combineTrustClientCount(fc))
+      combinedCount = fullCount.fold(Map.empty[String, Int])(fc => combineServicesClientCount(fc))
 
       groupCount = combinedCount.filter(m => taxServiceIds.contains(m._1))
     } yield groupCount
 
-  private val TRUSTS = "HMRC-TERS" // taxable "HMRC-TERS-ORG" and non taxable "HMRC-TERSNT-ORG"
-
-  private def combineTrustClientCount(count: Map[String, Int]): Map[String, Int] = {
-    val taxableTrustCounts = count.filter(m => m._1.contains(TRUSTS))
-    val combinedTrustCountValue = taxableTrustCounts.values.sum // total
-
-    if (combinedTrustCountValue > 0)
-      count.filterNot(m => m._1.contains(TRUSTS)) ++ Map(TRUSTS -> combinedTrustCountValue)
-    else count.filterNot(m => m._1.contains(TRUSTS)) // removes trusts if agent has no trust clients
+  /** Groups together client counts for services which act as 1 entity in the frontend but exist as 2 enrolment keys */
+  private def combineServicesClientCount(count: Map[String, Int]): Map[String, Int] = {
+    val TRUSTS = "HMRC-TERS" // taxable "HMRC-TERS-ORG" and non taxable "HMRC-TERSNT-ORG"
+    val CBC = "HMRC-CBC" // uk "HMRC-CBC-ORG" and non uk "HMRC-CBC-NONUK-ORG"
+
+    val trustCounts = count.filter(m => m._1.contains(TRUSTS))
+    val combinedTrustCountsValue = trustCounts.values.sum
```

Comment:
nothing wrong but you could replace both lines with
`val combinedTrustCountValue = count.keys.count(_.contains(TRUSTS))`
(and same below)

Reply:
Cool, can do 👍

Reply:
...actually, I'm not sure that's doing the same thing. there isn't an entry for each client. The key for a tax service will be there or not, and it's the value that contains the client count